jwst/master_background/expand_to_2d.py

```diff
@@ -0,0 +1,314 @@
+        background = bkg_for_IFUimage(input, tab_wavelength, tab_background)
+
+    else:
+        # Shouldn't get here.
+        raise RuntimeError("Input type {} is not supported."
+                           .format(type(input)))
+
+    return background
+
+
+def bkg_for_multislit(input, tab_wavelength, tab_background):
+    """Create a 2-D background for a MultiSlitModel.
+
+    Parameters
+    ----------
+    input : JWST MultiSlitModel
+        The input science data.
+
+    tab_wavelength : 1-D ndarray
+        The wavelength column read from the 1-D background table.
+
+    tab_background : 1-D ndarray
+        The flux column read from the 1-D background table, divided by
+        the npixels column.
+
+    Returns
+    -------
+    background : JWST MultiSlitModel
+        A copy of `input` but with the data replaced by the background,
+        "expanded" from 1-D to 2-D.
+    """
+
+    background = input.copy()
+
+    for (k, slit) in enumerate(input.slits):
+        wl_array = get_wavelengths(slit)
+        if wl_array is None:
+            raise RuntimeError("Can't determine wavelengths for {}"
+                               .format(type(slit)))
+
+        # Wherever the wavelength is NaN, the background flux should to be set
+        # to 0.  We replace NaN elements in wl_array with -1, so that np.interp
+        # will detect that those values are out of range (note the `left`
+        # argument to np.interp) and set the output to 0.
+        wl_array[np.isnan(wl_array)] = -1.
+
+        # bkg_flux will be a 2-D array, because wl_array is 2-D.
+        bkg_flux = np.interp(wl_array, tab_wavelength, tab_background,
+                             left=0., right=0.)
+
+        background.slits[k].data[:] = bkg_flux.copy()
+
+    return background
+
+
+def bkg_for_image(input, tab_wavelength, tab_background):
+    """Create a 2-D background for an ImageModel.
+
+    Parameters
+    ----------
+    input : JWST ImageModel
+        The input science data.
+
+    tab_wavelength : 1-D ndarray
+        The wavelength column read from the 1-D background table.
+
+    tab_background : 1-D ndarray
+        The flux column read from the 1-D background table, divided by
+        the npixels column.
+
+    Returns
+    -------
+    background : JWST ImageModel
+        A copy of `input` but with the data replaced by the background,
+        "expanded" from 1-D to 2-D.
+    """
+
+    background = input.copy()
+
+    wl_array = get_wavelengths(input)
+    if wl_array is None:
+        raise RuntimeError("Can't determine wavelengths for {}"
+                           .format(type(input)))
+
+    wl_array[np.isnan(wl_array)] = -1.
+
+    # bkg_flux will be a 2-D array, because wl_array is 2-D.
+    bkg_flux = np.interp(wl_array, tab_wavelength, tab_background,
+                         left=0., right=0.)
+
+    background.data[:] = bkg_flux.copy()
+
+    return background
+
+
+def bkg_for_IFUimage(input, tab_wavelength, tab_background):
```

Comment:
To appease the PEP8 gods, should be `bkg_for_ifuimage` or similar, since this is a function.

Reply:
It does pass flake8; however, using lower case does seem reasonable.  I'll change it to bkg_for_ifu_image.

Reply:
Cool.  Yeah, `flake8` doesn't do anything about function/class naming conventions in PEP8.  It ignores it, probably because there's legacy code in `python` itself that is not PEP8 compliant.  If you want that functionality in `flake8`, you can install `pep8-naming`, which will extend `flake8` to check for that. It's essentially a plugin.